hail/src/main/scala/is/hail/expr/ir/NestingDepth.scala

```diff
@@ -21,45 +39,78 @@ object NestingDepth {
 
     def computeBlockMatrix(bmir: BlockMatrixIR): Unit = computeChildren(bmir)
 
-    def computeIR(ir: IR, depth: Int): Unit = {
-      memo.bind(ir, depth)
+    def computeIR(ir: IR, depth: ScopedDepth): Unit = {
+      memo.bind(ir, depth.eval)
       ir match {
         case ArrayMap(a, name, body) =>
-          computeIR(a, depth)
-          computeIR(body, depth + 1)
+          computeIR(a, depth.incrementEval)
```

Comment:
I think this should still just be `depth`, right?

Reply:
oh, yes.